doc/source/whatsnew/v0.23.0.txt

```diff
@@ -658,6 +658,37 @@ Notice in the example above that the converted ``Categorical`` has retained ``or
 
 Note that the unintenional conversion of ``ordered`` discussed above did not arise in previous versions due to separate bugs that prevented ``astype`` from doing any type of category to category conversion (:issue:`10696`, :issue:`18593`).  These bugs have been fixed in this release, and motivated changing the default value of ``ordered``.
 
+.. _whatsnew_0230.api_breaking.pretty_printing:
+
+Better pretty-printing of DataFrames in a terminal
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Previously, the default value for the maximum number of columns was
+``pd.options.display.max_columns=20``. This meant that relatively wide data
+frames would not fit within the terminal width, and pandas would introduce line
+breaks to display these 20 columns. This resulted in an output that was
+relatively difficult to read:
+
+.. image:: _static/print_df_old.png
+
+If Python runs in a terminal, the maximum number of columns is now determined
+automatically so that the printed data frame fits within the current terminal
+width (``pd.options.display.max_columns=0``) (:issue:`17023`). If Python runs
+as a Jupyter kernel (such as the Jupyter QtConsole or a Jupyter notebook, as
+well as in many IDEs), this value cannot be inferred automatically and is thus
+set to `20` as in previous versions. In a terminal, this results in a much
+nicer output:
+
+.. image:: _static/print_df_new.png
+
+Note that if you don't like the new default, you can always set this option
+yourself. To revert to the old setting, you can run this line:
+
+.. code-block:: python
+
+  pd.options.display.max_columns = 20
+
+.. _whatsnew_0230.api:
```

Comment:
I think this one sneaked in due to merge conflict? (anyhow it can be removed)

Reply:
I'm sorry, what do you mean?

Reply:
According to the diff, you added this line. But this should not be added (therefore I assumed you added it by accident while updating against master with rebasing/merging). But so you can just remove this line.

Reply:
You mean line 690 (`.. _whatsnew_0230.api:`)?

Reply:
yes, that's the line on which I commented. There is no header following for which that link would make sense (there is actually already another link label on line 692)

Reply:
I thought there should be a section header before introducing the subsections. At least that's how it is done with `.. _whatsnew_0230.api_breaking:` in line 350 (but there is a heading after that). In any case, I'm happy to delete it.

Reply:
Yes, that is correct. But I don't understand the relation with this line? This link is just floating with no section or subsection header following it. You already have a header with link at line 661 - 664 ?

Reply:
It's for the section below:

```
.. _whatsnew_0230.api:

.. _whatsnew_0230.api.datetimelike:

Datetimelike API Changes
^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
But that header already has the "whatsnew_0230.api.datetimelike" label, it does not need two labels.

Reply:
OK, done.